### PR TITLE
Python code cleanup

### PR DESCRIPTION
**Pull request by ablakey**
This PR represents a significant step toward cleaning up the Python implementation of actionlib.  I focused on removing unused imports, removing semicolons, spacing, indentation.

I did not clean up line lengths.

I did not clean up any of the docstrings as I don't have experience with Doxygen and don't want to break anything.

### CODE REVIEW

**Comment by mikepurvis** [comment]
This seems reasonable to me— doesn't look like there are any significant public forks, so I don't see there being major merge issues with it. Is there development you're wanting to do on actionlib which is made easier by having the package style updated?

@mikaelarguedas What are your thoughts, as new maintainer?

**Comment by ablakey** [comment]
The goal is to give the package a more consistent style. I found a lot of the logic to be difficult to follow when studying the threading lock use, for example. Not trying to be a PEP8 pedant, but as far as I could tell, there was no consistent style.

More generally speaking, we're investigating opportunities to improve the library's behaviour when used in contexts with a lot of threading and/or inconsistent networking situations.

Hopefully this PR will make it easier to study and maintain actionlib's Python implementation to be worth the cost of a considerable hit to the blame history, mutation from the original, etc.

**Comment by mikaelarguedas** [comment]
It looks good to me. thanks @ablakey for homogenizing the style!
I'm surprised that the test has been labelled as flaky on the Kinetic buildfarm.
As soon as the tests pass it don't see any reason not to merge it.

File: src/actionlib/action_client.py [hunk 1]
```diff
@@ -475,12 +475,11 @@ def _get_live_statuses(self):
             live_statuses = filter(lambda x: x, live_statuses)
             return live_statuses
 
-
     ## Updates the statuses of all goals from the information in status_array.
     ##
     ## @param status_array (\c actionlib_msgs/GoalStatusArray)
     def update_statuses(self, status_array):
-        live_statuses = []
+        live_statuses = []  # TODO: unused local var.
```
Comment (dirk-thomas) [inline src/actionlib/action_client.py]: This line can safely be removed.

**Comment by ablakey** [comment]
@mikaelarguedas I'm not sure why the test isn't passing.  The tests for the Python implementation pass on my local machine. Is this something I need to explore further or an oddity of the buildfarm?

**Comment by mikaelarguedas** [comment]
given that the tests passed on jade for ubuntu 14.04 I don't think there is anything wrong on your side. We''l have to investigate on our hand to see why it's not passing on 16.04

**Comment by ablakey** [comment]
Are we in a holding pattern for something else to be resolved, or is there something we can do to move forward on this PR?

I'm prodding the unit tests that don't seem to pass, but I'm hesitant to base branches off branches of forks and go too far down that path. =)  Though this isn't explicitly blocking.

**Comment by mikaelarguedas** [comment]
Sorry because it was only a cleanup it went low on my priority list. I'll investigate the test failure on Kinetic shortly and will let you know once we're good to move forward.

**Comment by mikaelarguedas** [comment]
@ros-pull-request-builder retest this please

**Comment by mikaelarguedas** [comment]
It seems that it was due to a bug on the buildfarm which has been fixed since. Thanks for your work and your patience!

**Pull request merged**